deployment/terraform/lambda-functions/alert_sfn_failures/src/alert_sfn_failures.py

```diff
@@ -0,0 +1,53 @@
+import os
+import re
+import json
+import rollbar
+
+ECS_TASK_URL = "https://console.aws.amazon.com/ecs/home?region={region}#/clusters/{cluster}/tasks/{task}/details"
+
+environment = os.getenv('ENVIRONMENT', 'Staging')
+rollbar_token = os.getenv('ROLLBAR_SERVER_SIDE_ACCESS_TOKEN')
+rollbar.init(rollbar_token, environment)
+
+@rollbar.lambda_function
+def handler(event, context):
+    event = json.loads(event["Cause"])
```

Comment:
It looks like there some cases (step function run [here](https://eu-west-1.console.aws.amazon.com/states/home?region=eu-west-1#/executions/details/arn:aws:states:eu-west-1:249322298638:execution:stateMachineStagingAppCLI:a99e8ea3-008c-df2b-ea30-f58ba1d283c8_a5839bc8-b698-7163-2b24-721714e0dcec)) where `event["Cause"]` is an empty string, like when `"Error": "States.Timeout"`. This has been propagating to Rollbar, but as an uncaught exception (item [here](https://rollbar.com/OpenApparelRegistry/OpenApparelRegistry/items/630/)) rather than an intentionally submitted item.

My thoughts are:

1. The `check_api_limits` command seems to be timing out, not sure if that's something that just happens sometimes or if we should increase the timeout to see if that improves the success rate.
2. We should handle cases where the cause is an empty string, possibly by submitting the `Error` instead.

What are your thoughts here?

Reply:
Nice catch, I missed that. I added something that catches the `JSONDecodeError` and surfaces a generic message.

![image](https://user-images.githubusercontent.com/1774125/103579757-e21cad80-4ea6-11eb-9243-c91e8dbeda47.png)

I wanted to add more color here, like what task timed out, but as you can see here there really isn't much to go off of besides the name of the Step Functions error and the Lambda function itself.

https://rollbar.com/OpenApparelRegistry/OpenApparelRegistry/items/635/occurrences/147678787027/

Any ideas on how we can push more detail to the Lambda function? Maybe it's not necessary? I adjusted the timeout to 600 seconds as @jwalgran suggested.

Reply:
Some way to correlate the reported error back to the step function execution would be nice, but I like that Lambda isn't aware of the context that it's run in.

I can see an approach where we pass an envelope of arbitrary data to the Lambda reporting function to pass along to Rollbar, but I don't think that's a high priority at this time. Correlating recent failures and glancing at timestamps should be enough given the frequency that this is run at.